kernel/src/main/java/org/kframework/utils/BinaryLoader.java

```diff
@@ -96,12 +96,12 @@ public void saveOrDie(File file, Object o) {
 
     @Nullable
     public <T> T loadCache(Class<T> cls, File file) {
-        try (FileInputStream in = new FileInputStream(file)) {
-            return cls.cast(loadSynchronized(in));
+        try {
+            return cls.cast(loadSynchronized(file));
         } catch (FileNotFoundException e) {
             //ignored
         } catch (IOException | ClassNotFoundException e) {
-            kem.registerInternalHiddenWarning("Invalidating serialized cache due to corruption.", e);
+            kem.registerInternalWarning("Invalidating serialized cache due to corruption.", e);
```

Comment:
It did happen during the normal operation so far, precisely because of the bug this PR fixes. It took me a long time to figure out the root of the problem because this warning was hidden. From now on this shouldn't happen as part of normal operation unless definition was not kompiled with latest K. If this warning happens in future, we want to see it.

Do you know other scenarios why it might happen, unrelated to this fix or stale definition?

Reply:
definition was compiled with an old version of K /is/ normal operation.

Reply:
Fine, I reverted to avoid arguing. But this practice of hidden warnings needs rethinking. It is precisely because this warning was hidden that we didn't discover this bug until now.